proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java

```diff
@@ -58,11 +71,17 @@ public PointHandlerImpl(final String handle,
     this.blockedPointsPerBatch = blockedPointsPerBatch;
     this.prefix = prefix;
     String logPointsProperty = System.getProperty("wavefront.proxy.logpoints");
-    this.logPoints = logPointsProperty != null && logPointsProperty.equalsIgnoreCase("true");
+    this.logPointsFlag = logPointsProperty != null && logPointsProperty.equalsIgnoreCase("true");
 
     this.receivedPointLag = Metrics.newHistogram(new MetricName("points." + handle + ".received", "", "lag"));
 
     this.sendDataTasks = sendDataTasks;
+    this.pointHandlerExecutor =  Executors.newScheduledThreadPool(1,
```

Comment:
this seems a bit over-complicated (to take a thread to do this). perhaps just add a volatile long and check the loggable status if it's stale instead?